annotator-core/src/main/java/edu/ucr/cs/riple/core/registries/field/FieldRegistry.java

```diff
@@ -92,46 +93,63 @@ protected void setup() {
 
   @Override
   protected Builder<ClassFieldRecord> getBuilder() {
-    return values -> {
-      // Class flat name.
-      String clazz = values[0];
-      // Path to class.
-      Path path = Helper.deserializePath(values[1]);
-      CompilationUnit tree = Injector.parse(path);
-      if (tree == null) {
-        return null;
-      }
-      NodeList<BodyDeclaration<?>> members;
-      try {
-        members = Helper.getTypeDeclarationMembersByFlatName(tree, clazz);
-      } catch (TargetClassNotFound notFound) {
-        System.err.println(notFound.getMessage());
-        return null;
+    return new Builder<>() {
+      Pair<Path, CompilationUnit> lastParsedSourceFile = new Pair<>(null, null);
+
+      @Override
+      public ClassFieldRecord build(String[] values) {
+        // This method is called with values in format of: [class flat name, path to source file].
+        // To avoid parsing a source file multiple times, we keep the last parsed
+        // source file in a reference.
+        // This optimization is according to the assumption that Scanner
+        // visits all classes within a single compilation unit tree consecutively.
+        // Path to class.
+        Path path = Helper.deserializePath(values[1]);
+        CompilationUnit tree;
+        if (lastParsedSourceFile.a != null && lastParsedSourceFile.a.equals(path)) {
+          // Already visited.
+          tree = lastParsedSourceFile.b;
+        } else {
+          tree = Injector.parse(path);
+        }
+        lastParsedSourceFile = new Pair<>(path, tree);
```

Comment:
Could this line be inside the preceding `else` block?

Reply:
cfcfb35d70698ee0514e01e9faf0683df9a84872